### PR TITLE
Xml extensions

### PR DESCRIPTION
**Pull request by schaefi**
An initiative to allow our users to extend the KIWI image description, and a start to think about a plugin interface. So far kiwi has such an interface for commandline tasks, thus adding new commands can be done, but we might also want to allow to hook into other parts of the KIWI processing. So this is still an open discussion and a start

Thanks

### CODE REVIEW

**Comment by schaefi** [comment]
Note:

* xml_parse.py
* kiwi.rng

is auto generated code

File: doc/source/extensions.rst [hunk 1]
```diff
@@ -0,0 +1,146 @@
+Extending KIWI with custom operations
```
Comment (tomschr) [inline doc/source/extensions.rst]: I would use title style here: "Extending KIWI With Custom Operations"

File: doc/source/extensions.rst [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+   provide a way to embed custom information which is
+   outside of the scope of the general schema as it is
+   provided by KIWI today.
+
+   This document describes how to create an extension plugin
+   for the KIWI schema in order to allow to add and validate
+   additional information in the KIWI image description.
+
+   Such a schema extension can be used in an additional KIWI
+   task plugin in order to provide a new subcommand for KIWI.
```
Comment (tomschr) [inline doc/source/extensions.rst]: Same here, remove "in order"

File: doc/source/extensions.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+   As of today there is no other plugin interface except for
+   providing additional KIWI commands implemented.
+
+   Depending on the demand for custom plugins, the interface
+   to hook in code into other parts of the KIWI processing
+   needs to be extended.
+
+   This description applies for version |version|.
+
+The <extension> section
```
Comment (tomschr) [inline doc/source/extensions.rst]: Use title style: `The <extension> Section`

File: doc/source/extensions.rst [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                <my_plugin:title name="cool stuff"/>
+            </my_plugin:my_feature>
+        </extension>
+    </image>
+
+* Any toplevel namespace must exist only once
+* Multiple different toplevel namespaces are allowed,
+  e.g my_plugin_a, my_plugin_b
+
+RelaxNG Schema for the Extension
```
Comment (tomschr) [inline doc/source/extensions.rst]: RelaxNG -> RELAX NG

File: doc/source/extensions.rst [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+If an extension section is found, KIWI looks up its namespace and asks
+the main XML catalog for the schema file to validate the extension data.
+The schema file must be a RelaxNG schema in the .rng format. We recommend
+to place the schema as :file:`/usr/share/xml/kiwi/my_plugin.rng`
+
+For the above example the RelaxNG Schema in the compressed format
+:file:`my_plugin.rnc` would look like this:
+
+.. code:: bash
```
Comment (tomschr) [inline doc/source/extensions.rst]: It's pity that there is no syntax highlighting for RNC.

File: doc/source/extensions.rst [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+    namespace rng = "http://relaxng.org/ns/structure/1.0"
```
Comment (tomschr) [inline doc/source/extensions.rst]: I would remove the rnc namespace as it isn't really used in that example.

File: doc/source/extensions.rst [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+    </catalog>
+
+For resolving the catalog KIWI uses the :command:`xmlcatalog` command
+and the main XML catalog from the system which is :file:`/etc/xml/catalog`.
+
+.. note::
+
+    It depends on the distribution and its version how the main catalog
+    gets informed about the existence of the KIWI extension catalog file.
+    Please consult the distribution manual about adding XML catalogs
```
Comment (tomschr) [inline doc/source/extensions.rst]: Great note! Just add the missing point at the end of the sentence. :-)

**Comment by schaefi** [comment]
Thanks Tom for the review, all updated :)

**Pull request merged**